Fix Bug 1679654 - Add ability to reset machinery search field

abowler2 opened the pull request:
The ability to reset the Machinery Search input to its original state is not currently intuitive (user must clear the input and hit enter). 

This PR follows the [specification for Concordance Search](https://github.com/mozilla/pontoon/blob/master/specs/0106-concordance-search.md) and switches the 'search' icon to a 'close' button (`x`) when a search is made. When the 'close' button is clicked the search input will be cleared and returned to its original state.

This does not currently work for the FTL files due to [Bug 1676750](https://bugzilla.mozilla.org/show_bug.cgi?id=1676750). The input is cleared but the original state is not restored.

mathjazz commented on frontend/src/modules/machinery/components/Machinery.css:
```diff
@@ -23,6 +23,12 @@
+.machinery > .search-wrapper > label > button {
+    background-color: transparent;
+    border: none;
+    color: #aaaaaa;
```
Seems like `<button>` gets some padding assigned (although I don't know from where?), which moves the times icon towards left compared to the magnifier icon.

Adding `padding: 0 4px;` here should fix the unexpected jump.

mathjazz commented on frontend/src/modules/machinery/components/Machinery.js:
```diff
@@ -54,6 +54,12 @@ export default class Machinery extends React.Component<Props> {
+    handleResetSearch = (event: SyntheticMouseEvent<>) => {
+        event.preventDefault();
+        this.searchInput.current.value = '';
+        this.props.searchMachinery(this.searchInput.current.value);
```
I'd expect `this.props.searchMachinery('');` to be sufficient here, because `componentDidUpdate` should set `this.searchInput.current.value` accordingly.

And I don't think we need to prevent default?

abowler2 replied:
Yep, much cleaner to just set with an empty string.

I wasn't sure about the prevent default either. I added it to protect against an action that I might have overlooked, but it does not appear to be needed.

mathjazz reviewed (commented):
Great job!

Works like a charm.

Just two nits / questions I found along the way.

abowler2 commented:
I just noticed that when switching tabs the search term still stays in the form as it should but the 'close' button is no longer present.  I will work on fixing that.

mathjazz commented on frontend/src/modules/machinery/components/Machinery.js:
```diff
@@ -72,12 +77,23 @@ export default class Machinery extends React.Component<Props> {
-                        <div className='fa fa-search'></div>
-                    </label>
+                    {showResetButton ? (
+                        <label htmlFor='machinery-search'>
```
Looks like the `<label>` element is the same, so you can move the condition inside of it.

mathjazz reviewed (commented):
Thanks for the update and spotting the issue the review hasn't. :-)

A test would be nice.

abowler2 commented:
Updated based on comments and added a test.

Not sure what is happening with the CI tests though. Locally the tests pass. I rebased to ensure I had the latest changes to `master` but it is still failing unrelated to my changes.

mathjazz reviewed (approved):
Great job!

That Travis error is unrelated, we can tackle it separately if it resurfaces.
